サーバーに設定する

はらぺこ鯖に設定してみる。上手く動かない部分があればバグ修正する。

色々とバグ修正。で、デバッグモードで動かせるようになったが、ログインした時点で以下のエラーが発生…。

```
[ERROR] debug: Not handled error was raised.
Traceback (most recent call last):
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "tmpbox.py", line 152, in page_index
    login_session, _ = verify_login_session()
  File "tmpbox.py", line 81, in verify_login_session
    login_session = db.check_login_session(session.get('token', None))
  File "./tmpbox_db_accessor.py", line 584, in check_login_session
    return self.session_scope(
  File "./tmpbox_db_accessor.py", line 393, in session_scope
    result = query_func(session)
  File "./tmpbox_db_accessor.py", line 585, in <lambda>
    lambda s: self.__session_check_login_session(s, session_id), True)
  File "./tmpbox_db_accessor.py", line 596, in __session_check_login_session
    login_session = SessionState.filter_check_expires(
  File "./tmpbox_db_accessor.py", line 145, in filter_check_expires
    func.timestampdiff(text('minute'), SessionState.access_dt, functions.now())
NameError: name 'func' is not defined
Traceback (most recent call last):
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 1953, in full_dispatch_request
    return self.finalize_request(rv)
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 1968, in finalize_request
    response = self.make_response(rv)
  File "/home/murachi/.local/share/virtualenvs/tmpbox-uUfjWWGP/lib/python3.8/site-packages/flask/app.py", line 2127, in make_response
    raise TypeError(
TypeError: The view function did not return a valid response. The return type must be a string, dict, tuple, Response instance, or WSGI callable, but it was a NameError.
[pid: 25451|app: 0|req: 10/10] 61.125.220.180 () {40 vars in 736 bytes} [Mon May 11 19:14:04 2020] GET / => generated 0 bytes in 8 msecs (HTTP/1.1 500) 0 headers in 0 bytes (0 switches on core 0)
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,5 @@
 __pycache__/
+.venv/
 /src/static/*.css
 /src/static/*.css.map
 /src/static/*.js

--- a/setup.sh
+++ b/setup.sh
@@ -1,0 +1,4 @@
+#!/bin/bash
+export PIPENV_VENV_IN_PROJECT=1
+pipenv sync
+sudo pipenv run python setup.py

--- a/src/run.sh
+++ b/src/run.sh
@@ -1,4 +1,7 @@
 #!/bin/bash
-export FLASK_APP=tmpbox.py
-export FLASK_ENV=development
-flask run
+if [ ${EUID:-${UID}} != 0 ]; then
+  echo 'Cannot run without superuser. Please call me with sudo, or after change to superuser.'
+  exit 1
+fi
+export PIPENV_VENV_IN_PROJECT=1
+pipenv run uwsgi --ini conf.d/uwsgi-debug.ini

